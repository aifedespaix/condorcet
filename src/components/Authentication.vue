<template>
  <div>
    <div class="background"></div>
    <section class="Authentication">
      <b-message title="Connexion" :closable="false">
        <div class="has-text-right">
          <button class="button" @click="signIn">
            <span class="icon is-small"> <i class="fab fa-google"></i> </span
            ><span> Google </span>
          </button>
        </div>
      </b-message>
    </section>
  </div>
</template>

<script>
import { mapActions } from "vuex";
import { signIn } from "../utils/authentication.js";

export default {
  name: "Authentication",
  methods: {
    ...mapActions("user", ["setUser"]),
    ...mapActions("app", ["addNotification"]),
    signIn() {
      signIn(user => {
        this.setUser(user);

        this.$router.push("polls");
        this.addNotification({ text: "Vous êtes connecté", status: "info" });
      });
    }
  }
};
</script>

<style scoped>
.background {
  position: fixed;
  top: 0;
  left: 0;
  z-index: -1000;
  background: url("../assets/images/scrutin2.jpg");
  filter: blur(10px);
  background-size: cover;
  width: 100%;
  height: 100vh;
}
.Authentication {
  position: absolute;
  top: 50%;
  right: 50%;
  transform: translate(50%, -50%);
  background: rgba(100, 100, 100, 0.8);
  min-width: 300px;
  border-radius: 4px;
}
.Authentication__Google {
  height: 50px;
}
</style>
