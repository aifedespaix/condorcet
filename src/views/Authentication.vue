<template>
  <v-jumbotron color="grey lighten-2">
    <v-container fill-height>
      <v-layout align-center>
        <v-flex>
          <h3 class="display-3">Bienvenue sur Condorcet</h3>
          <span class="subheading"
            >Condorcet est une application permettant de réaliser des scrutin en
            utilisant la méthode de Condorcet</span
          >
          <v-divider class="my-3"></v-divider>
          <div class="title mb-3">Accéder à l'application :</div>
          <v-btn large color="primary" class="mx-0" @click="signIn('google')"
            >Google</v-btn
          >
          <v-btn large color="primary" class="mx-0" @click="signIn('github')"
            >Github</v-btn
          >
        </v-flex>
      </v-layout>
    </v-container>
  </v-jumbotron>
</template>

<script>
import { mapActions } from "vuex";
import { signIn } from "../utils/authentication.js";

export default {
  name: "Authentication",
  methods: {
    ...mapActions("user", ["setUser"]),
    ...mapActions("app", ["addNotification"]),
    signIn(provider) {
      signIn(
        provider,
        user => {
          this.setUser(user);

          this.$router.push({ name: "polls_list" });
          this.addNotification({ text: "Vous êtes connecté", status: "info" });
        },
        error => {
          switch (error.code) {
            case "auth/account-exists-with-different-credential":
              this.addNotification({
                text: "Veuillez vous identifier via Google",
                status: "error"
              });
              break;

            default:
              break;
          }
        }
      );
    }
  }
};
</script>

<style scoped>
.background {
  position: fixed;
  top: 0;
  left: 0;
  z-index: -1000;
  background: url("../assets/images/scrutin2.jpg");
  filter: blur(10px);
  background-size: cover;
  width: 100%;
  height: 100vh;
}
.Authentication {
  position: absolute;
  top: 50%;
  right: 50%;
  transform: translate(50%, -50%);
  background: rgba(100, 100, 100, 0.8);
  min-width: 300px;
  border-radius: 4px;
}
.Authentication__Google {
  height: 50px;
}
</style>
