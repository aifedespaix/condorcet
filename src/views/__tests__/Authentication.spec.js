import { shallowMount } from "@vue/test-utils";
import Authentication from "../Authentication.vue";

describe("Authentication.vue", () => {
  it("should render", () => {
    const wrapper = shallowMount(Authentication);

    expect(wrapper).toMatchSnapshot();
  });
});
