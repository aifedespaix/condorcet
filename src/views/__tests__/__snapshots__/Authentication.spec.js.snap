// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication.vue should render 1`] = `
<v-jumbotron-stub color="grey lighten-2" tag="div" height="400px">
  <v-container-stub tag="div" fill-height="">
    <v-layout-stub tag="div" align-center="">
      <v-flex-stub tag="div">
        <h3 class="display-3">Bienvenue sur Condorcet</h3> <span class="subheading">Condorcet est une application permettant de réaliser des scrutin en
          utilisant la méthode de Condorcet</span>
        <v-divider-stub class="my-3"></v-divider-stub>
        <div class="title mb-3">Accéder à l'application :</div>
        <v-btn-stub color="primary" activeclass="v-btn--active" tag="button" large="true" type="button" class="mx-0">Google</v-btn-stub>
        <v-btn-stub color="primary" activeclass="v-btn--active" tag="button" large="true" type="button" class="mx-0">Github</v-btn-stub>
      </v-flex-stub>
    </v-layout-stub>
  </v-container-stub>
</v-jumbotron-stub>
`;
